src/bottomSheet/BottomSheet.js

```diff
@@ -16,34 +17,57 @@ class BottomSheet extends Component {
     super(props);
     this.state = {
       listHeight: undefined,
+      isVisible: this.props.isVisible,
     };
   }
 
+  componentDidUpdate(prevProps) {
+    if (prevProps.isVisible !== this.props.isVisible) {
+      this.setState({ isVisible: this.props.isVisible });
+    }
+  }
+
+  hideModal = () => this.setState({ isVisible: false });
+
   onLayout = (event) =>
     this.setState({ listHeight: event.nativeEvent.layout.height });
 
-  render() {
+  renderChildren = () => {
+    const { children } = this.props;
     const { listHeight } = this.state;
-    const { isVisible, modalProps, children } = this.props;
     const maxHeight = listHeight < MAX_HEIGHT ? listHeight : MAX_HEIGHT;
 
+    return (
+      <SafeAreaView style={styles.safeAreaView}>
+        <View style={styles.modalView}>
+          <View
+            style={([styles.listContainer], { maxHeight })}
+            onLayout={this.onLayout}
+          >
+            <ScrollView>{children}</ScrollView>
+          </View>
+        </View>
+      </SafeAreaView>
+    );
+  };
+
+  render() {
+    const { modalProps, isCancelable } = this.props;
+    const { isVisible } = this.state;
     return (
       <Modal
         animationType="slide"
         transparent={true}
         visible={isVisible}
         {...modalProps}
       >
-        <SafeAreaView style={styles.safeAreaView}>
-          <View style={styles.modalView}>
-            <View
-              style={([styles.listContainer], { maxHeight })}
-              onLayout={this.onLayout}
-            >
-              <ScrollView>{children}</ScrollView>
-            </View>
-          </View>
-        </SafeAreaView>
+        {isCancelable === true ? (
```

Comment:
Where is `isCancelable` coming from? You need to make updates to the docs if you're adding a new prop and an update to `index.d.ts`